### PR TITLE
volume profiling changes for cluster

### PR DESCRIPTION
**Pull request by gnehapk**
tendrl-github-id: #879 

Signed-off-by: Neha Gupta <gnehapk@gmail.com>

### CODE REVIEW

File: src/modules/clusters/cluster-list/cluster-list.html [hunk 1]
```diff
@@ -49,28 +49,26 @@ <h1>No Clusters Detected</h1>
...
                                     <i class="glyphicon glyphicon-refresh glyphicon-refresh-animate" data-ng-class="{'hidden': clusterCntrl.disProfilingBtnClicked !== true }"></i>
                                 </a>
-                                <a ng-class="{'disable-pointer': $root.userRole === 'limited' || clusterCntrl.enProfilingBtnClicked}" ng-click="clusterCntrl.doProfilingAction($event, cluster, 'Enable', cluster.clusterId)" uib-tooltip="Enable Profiling" ng-show="cluster.isProfilingEnabled === 'Disabled' || cluster.isProfilingEnabled === 'Mixed'">Enable Profiling
+                                <a ng-class="{'disable-pointer': $root.userRole === 'limited' || cluster.disableAction || (cluster.jobType === 'EnableDisableVolumeProfiling' && cluster.currentStatus === 'in_progress')}" ng-click="clusterCntrl.doProfilingAction($event, cluster, 'Enable', cluster.clusterId)" uib-tooltip="Enable Profiling" ng-show="cluster.isProfilingEnabled === 'Disabled' || cluster.isProfilingEnabled === 'Mixed'">Enable Profiling
```
Comment (cloudbehl) [inline src/modules/clusters/cluster-list/cluster-list.html:62]: can you call a function, it is difficult for someone to understand

$root.userRole === 'limited' || cluster.disableAction || (cluster.jobType === 'EnableDisableVolumeProfiling' && cluster.currentStatus === 'in_progress')

File: src/modules/clusters/cluster-list/cluster-list.html [hunk 2]
```diff
@@ -110,12 +108,12 @@ <h5 class="pull-left counts">{{cluster.alertCount}}</h5>
...
                                 <p uib-tooltip-template="'../modules/clusters/cluster-list/unmanage-cluster-tooltip.html'" tooltip-enable="clusterCntrl.isTooltipEnable(cluster.message)" tooltip-placement="auto">{{cluster.disableExpand ? "Expanding cluster" : cluster.message}}</p>
-                                <!-- <p><a ng-click="clusterCntrl.openErrorModal(cluster)" ng-show="cluster.currentStatus === 'failed'" class="details-link">View Details</a></p> -->
-                                <p><a ng-click="clusterCntrl.goToTaskDetail(cluster)" ng-show="cluster.currentStatus === 'in_progress' || cluster.currentStatus === 'failed'" class="details-link">View Details</a></p>
+                                <p><a ng-click="clusterCntrl.goToTaskDetail(cluster)" ng-show="(cluster.currentStatus === 'in_progress' && cluster.jobType !== 'EnableDisableVolumeProfiling')|| cluster.currentStatus === 'failed'" class="details-link">View Details</a></p>
```
Comment (cloudbehl) [inline src/modules/clusters/cluster-list/cluster-list.html]: same as above

File: src/commons/stores/cluster-store.js [hunk 3]
```diff
@@ -296,10 +292,27 @@
...
+
+        function _getProfileStatus(temp, cluster) {
+            var status = "",
+                profileStatus = {
+                    "enabled": "Enabled",
+                    "disabled": "Disabled",
+                    "mixed": "Mixed"
+                };
+
+            if (temp.jobType === "EnableDisableVolumeProfiling" && temp.currentStatus === "in_progress") {
```
Comment (cloudbehl) [inline src/commons/stores/cluster-store.js:308]: Is there a chance that some other jobtype can come? why is this check 'temp.jobType === "EnableDisableVolumeProfiling" '

**Review by cloudbehl — CHANGES_REQUESTED** [review]

File: src/commons/stores/cluster-store.js (hunk 3 above)
Comment (gnehapk) [reply src/commons/stores/cluster-store.js:308]: @cloudbehl Yes. There are various jobs that can be in `current_job` like ImportCluster, UnmanageCluster.

File: src/modules/clusters/cluster-list/cluster-list.html (hunk 1 above)
Comment (gnehapk) [reply src/modules/clusters/cluster-list/cluster-list.html:62]: @cloudbehl Calling a function was not doing the 2 way binding thing. Hence, need to do this way.

File: src/modules/clusters/cluster-list/cluster-list.html (hunk 2 above)
Comment (gnehapk) [reply src/modules/clusters/cluster-list/cluster-list.html]: @cloudbehl This can be done via function.

**Review by a2batic — APPROVED** [review]

**Review by cloudbehl — APPROVED** [review]
LGTM

**Review by r0h4n — APPROVED** [review]

**Pull request merged**